### PR TITLE
Add notebook with miscellaneous Datalab workflows

### PR DESCRIPTION
**Pull request by elisno**
Tutorials added
- [x] Check for null values
- [x] Find underperforming groups (by clustering with kmeans)
- [x] Perform data valuation
- [x] Pre-compute knn graph (with Faiss)
  - ~~Still WIP, I believe it has some numerical instability.~~ Not a priority when using third party library for constructing a knn graph.
- [x] Non-IID
- [x] ~~Class imbalance~~

Remaining TODOs:

- [x] Add dependencies for running tutorial
  - Faiss, seaborn?
- [x] Link to other resources
   - Issue type descriptions
- [x] Review non-default issue-type params for each section

**Screenshots**

The notebook hasn't been added to the docs yet in this PR, but here's a scroll over the docs:

https://github.com/cleanlab/cleanlab/assets/18127060/06c37fce-9c89-43c3-93a4-7c90ed4c7d91

### CODE REVIEW

**Comment by jwmueller** [comment]
closes https://github.com/cleanlab/cleanlab/issues/920

File: docs/source/tutorials/datalab/index.rst
Comment (jwmueller) [inline docs/source/tutorials/datalab/index.rst:14]:
```suggestion
   Miscellaneous Workflows <workflows>
```

**Review by jwmueller — APPROVED** [review]
Looks good, but please make the following changes before you merge (no need for another review):

1. Order these sections by how useful they are:
Approximate KNN, Data Valuation, Underperforming Groups with your own provided clusters, Non-IID, Null values, Class Imbalance

If you can add one sentence to the Null values / Class imbalance sections indicating what is useful in these sections (they don't seem that useful to me, these issue types are auto-detected by default, so I guess these sections are just helping user understand these issue types).

Finally, what happened to showing underperforming groups based on data slices along a categorical feature of a tabular dataset?  Can you open a separate PR to demonstrate that as another application of providing your own 'clusters' (in this case simply the levels of the categorical feature, eg: "California", "Nevada", "Washington", ...)

**Comment by jwmueller** [comment]
also I pushed many changes so definitely review those before merge

**Comment by elisno** [comment]
> Finally, what happened to showing underperforming groups based on data slices along a categorical feature of a tabular dataset? Can you open a separate PR to demonstrate that as another application of providing your own 'clusters' (in this case simply the levels of the categorical feature, eg: "California", "Nevada", "Washington", ...)

I'll open a follow-up PR, adding a similar section for data slices along a categorical feature of a tabular dataset.

**Pull request merged**
